packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -0,0 +1,373 @@
+  ///
+  /// Typically these actions can represent additional modes of searching
+  /// (like voice search), an avatar, a separate high-level action (such as
+  /// current location) or an overflow menu. There should not be more than
+  /// two trailing actions.
+  final Iterable<Widget>? trailing;
+
+  /// Called when the user taps this search bar.
+  final GestureTapCallback? onTap;
+
+  /// Invoked upon user input.
+  final ValueChanged<String>? onChanged;
+
+  /// Optional size constraints for the search bar.
+  ///
+  /// If null, the value of [SearchBarThemeData.constraints] will be used. If
+  /// this is also null, then the constraints defaults to:
+  /// ```dart
+  /// const BoxConstraints(minWidth: 360.0, maxWidth: 800.0, minHeight: 56.0)
+  /// ```
+  final BoxConstraints? constraints;
+
+  /// The elevation of the search bar's [Material].
+  ///
+  /// If null, the value of [SearchBarThemeData.elevation] will be used. If this
+  /// is also null, then default value is 8.0.
+  final MaterialStateProperty<double?>? elevation;
+
+  /// The search bar's background fill color.
+  ///
+  /// If null, the value of [SearchBarThemeData.backgroundColor] will be used.
+  /// If this is also null, then the default value is [ColorScheme.surface].
+  final MaterialStateProperty<Color?>? backgroundColor;
+
+  /// The shadow color of the search bar's [Material].
+  ///
+  /// If null, the value of [SearchBarThemeData.shadowColor] will be used.
+  /// If this is also null, then the default value is [ColorScheme.shadow].
+  final MaterialStateProperty<Color?>? shadowColor;
+
+  /// The surface tint color of the button's [Material].
+  ///
+  /// See [Material.surfaceTintColor] for more details.
+  ///
+  /// If null, the value of [SearchBarThemeData.surfaceTintColor] will be used.
+  /// If this is also null, then the default value is [ColorScheme.surfaceTint].
+  final MaterialStateProperty<Color?>? surfaceTintColor;
+
+  /// The highlight color that's typically used to indicate that
+  /// the button is focused, hovered, or pressed.
+  final MaterialStateProperty<Color?>? overlayColor;
+
+  /// The color and weight of the search bar's outline.
+  ///
+  /// This value is combined with [shape] to create a shape decorated
+  /// with an outline.
+  ///
+  /// If null, the value of [SearchBarThemeData.side] will be used. If this is
+  /// also null, the search bar doesn't have a side by default.
+  final MaterialStateProperty<BorderSide?>? side;
+
+  /// The shape of the search bar's underlying [Material].
+  ///
+  /// This shape is combined with [side] to create a shape decorated
+  /// with an outline.
+  ///
+  /// If null, the value of [SearchBarThemeData.shape] will be used.
+  /// If this is also null, then the default value is 16.0 horizontally.
+  /// Defaults to [StadiumBorder].
+  final MaterialStateProperty<OutlinedBorder?>? shape;
+
+  /// The padding between the search bar's boundary and its child.
```

Comment:
```suggestion
  /// The padding between the search bar's boundary and its contents.
```

Since search bar doesn't have an official `child`.